src/tribler/core/components/session.py

```diff
@@ -27,7 +27,7 @@ class Session:
     def __init__(self, config: TriblerConfig = None, components: List[Component] = (),
                  shutdown_event: Event = None, notifier: Notifier = None, failfast: bool = True):
         # deepcode ignore unguarded~next~call: not necessary to catch StopIteration on infinite iterator
-        self.exit_code = 0
+        self.exit_code = None
```

Comment:
It seems that it is possible to return `None` instead of `0` here: https://github.com/kozlovsky/tribler/blob/process_locker/src/tribler/core/start_core.py#L120

Reply:
I'm not sure I understand. In PR it already returns `None` and not `0`. Or you mean to replace `return session.exit_code` with `return session.exit_code or None`?

Reply:
To me it looks clearer is the `exit_code` is initially set to None, with the meaning `the exit code is not set`, comparing to converting `0` to `None` at a latter stage.